src/test/kotlin/uk/gov/dluhc/printapi/database/repository/AnonymousElectorDocumentSummaryRepositoryIntegrationTest.kt

```diff
@@ -0,0 +1,222 @@
+            val aed2ApplicationReference = aValidApplicationReference()
+            val aed2FirstCertificate = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(10),
+                requestDateTime = currentDateTimeInstant.minusSeconds(10),
+                contactDetails = buildAedContactDetails(surname = "Z")
+            )
+            val aed2SecondCertificate = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(9),
+                requestDateTime = currentDateTimeInstant.minusSeconds(8),
+                contactDetails = aed2FirstCertificate.contactDetails!!
+            )
+            val aed2LatestCertificate = buildAnonymousElectorDocument(
+                gssCode = gssCode,
+                sourceReference = aed2SourceReference,
+                applicationReference = aed2ApplicationReference,
+                issueDate = currentDate.minusDays(9),
+                requestDateTime = currentDateTimeInstant, // View will return this latest record
+                contactDetails = aed2FirstCertificate.contactDetails!!
+            )
+            val aed3 = buildAnonymousElectorDocument(gssCode = gssCode, issueDate = currentDate.plusDays(10))
+            val aed4 = buildAnonymousElectorDocument(gssCode = gssCode, issueDate = currentDate)
+            val otherAed = buildAnonymousElectorDocument(gssCode = anotherGssCode())
+
+            anonymousElectorDocumentRepository.saveAll(
+                listOf(
+                    aed1FirstCertificate, aed1SecondCertificate, aed1LatestCertificate,
+                    aed2FirstCertificate, aed2SecondCertificate, aed2LatestCertificate,
+                    aed3, aed4, otherAed
+                )
+            )
+
+            val expectedSummaryRecord1 = buildAnonymousElectorDocumentSummaryViewFromAedEntity(aed3)
+            val expectedSummaryRecord2 = buildAnonymousElectorDocumentSummaryViewFromAedEntity(aed4)
+            val expectedSummaryRecord3 = buildAnonymousElectorDocumentSummaryViewFromAedEntity(aed1LatestCertificate)
+            val expectedSummaryRecord4 = buildAnonymousElectorDocumentSummaryViewFromAedEntity(aed2LatestCertificate)
+            val otherAedSummaryRecord = buildAnonymousElectorDocumentSummaryViewFromAedEntity(otherAed)
+
+            // When
+            val actual = anonymousElectorDocumentSummaryRepository
+                .findAllByGssCodeInAndSourceType(
+                    gssCodes = listOf(gssCode),
+                    sourceType = ANONYMOUS_ELECTOR_DOCUMENT,
+                    pageRequest = withPageRequestAndSortOrder(0, 100)
```

Comment:
Ok, so we're testing some pagination behaviour, despite the fact it's not supported via the API yet?

I wonder if this is trying to test too many concerns and could be split up a bit?

Reply:
The repository requires pageRequest, so added this test.